### ISSUE TITLE
Upgrade of package install ALL available updates.

### ISSUE THREAD

**Issue by poetaster** [sailfishos-chum/sailfishos-chum-gui#67]
This is undesirable. I had chosen a package with a new version, pulled down update and all packages were installed.

**Comment by rinigus** [comment]
Seems to be PackageKit issue. Please check with pkcon and file an issue in PackageKit repo. Would be great to have a reference to that issue here as well.

**Comment by Olf0** [comment]
Although this is a "duplicate" of https://github.com/mentaljam/harbour-storeman/issues/96 , Rinigus' [comment advises the right thing to do](https://github.com/sailfishos-chum/sailfishos-chum-gui/issues/67#issuecomment-1006743143).

**Comment by poetaster** [comment]
Our version of PackageKit is not current, if I'm not mistaken. we are at 1.1.13 and current is 1.2.5. According to current documentation: 
`Update the system by installing available updates. If a list of packages is specified, only install updates for these packages.`

I don't have any means to test this but hesitate to file a bug report there.

**Comment by Olf0** [comment]
> Our version of PackageKit is not current, if I'm not mistaken. we are at 1.1.13 and current is 1.2.5. 

"Our" = on SailfishOS?   You can take a look at PackageKit on many Linux distributions, it is installed by default on ?buntu, should be available for Debian and may be available for OpenSUSE and Fedora / RHEL (well, `zypper`, `dnf` are so much nicer, but front-ends only for RPM), too.

> According to current documentation: `Update the system by installing available updates. If a list of packages is specified, only install updates for these packages.`

Which and where?  Can you please provide a link to this quote.

For me it sounds like a quote from `pkcon` man-page for `update [PACKAGES]`.  This is merely the front-end.  I assume that this bug is rather in `libpackagekit` etc., the "middle-ware" between `pkcon` and the various supported back-ends (`libzypp` being one of them and IIRC the one used for Sailfish OS).

> I don't have any means to test this but hesitate to file a bug report there.

Maybe there is already one?  I have never searched, because a `libsolv0` bug bugs me more.

**Comment by Olf0** [comment]
> Ah, it just occured to me, die you guys mean upstream https://github.com/sailfishos/PackageKit-Qt ?

No, definitely not that one, and also not the GTK one, but the CLI front-end `pkcon` and the "middle-ware" `libpackagekit` at IIRC freedesktop.org.

**Comment by Olf0** [comment]
Note that "**Upgrade of one package install ALL available updates.**" is not always the case, at least not with `pkcon` and Storeman.

**Comment by poetaster** [comment]
> Which and where? Can you please provide a link to this quote.
> 
> For me it sounds like a quote from `pkcon` man-page for `update [PACKAGES]`. This is merely the front-end. I assume that this bug is rather in `libpackagekit` etc., the "middle-ware" between `pkcon` and the various supported back-ends (`libzypp` being one of them and IIRC the one used for Sailfish OS)

I'll look into this but first: libpackagekit

**Comment by poetaster** [comment]
> Maybe there is already one?

I looked. Apparently not. But I may have missed it and will revisit.

> I have never searched, because a certain `libsolv0` bug bugs me more, and I fail to file it for > 1,5 years.

I'll keep an eye out for this in parallel.

**Comment by poetaster** [comment]
> Note that "**Upgrade of one package install ALL available updates.**" is not always the case, at least not with `pkcon` and Storeman.

Yes. Now it's a pitty we don't have that source. However, maybe it's time for a forum question?

**Comment by Olf0** [comment]
> > I have never searched, because a certain `libsolv0` bug bugs me more, and I fail to file it for > 1,5 years.
> 
> I'll keep an eye out for this in parallel.

???
You do not know what it is yet, right?
[It was recently documented at FSO](https://forum.sailfishos.org/t/release-notes-suomenlinna-4-3-0/8495/243?u=olf), but originally for Storeman: https://github.com/mentaljam/harbour-storeman/issues/108#issuecomment-886225570
I am pretty sure it is `libsolv0`'s fault by logic (it is the component performing the resolving), but have not technically traced it to `libsolv0` (and may never do, with or without getting around to report it).

**Comment by poetaster** [comment]
> ???
> You do not know what it is yet, right?

NO. And the rabbit hole yeilds only, more hole :) But that's beside the point. I meant that I would look up libsolv0 and look up issue in the sense of research. Frankly, I think this is one of those 'things' that 'just goes away'. But that's what I say about anything I don't entirely understand.

**Comment by Olf0** [comment]
> […] Frankly, I think this is one of those 'things' that 'just goes away'. But that's what I say about anything I don't entirely understand.

This hope is also the approach of the lazy; me in this case.
Didn't work within the last three years, unfortunately; I suppose I have not been waiting long enough, yet. :wink:
